precompiles/staking/types.go

```diff
@@ -149,6 +155,69 @@ func NewMsgCreateValidator(args []interface{}, denom string) (*stakingtypes.MsgC
 	return msg, delegatorAddress, nil
 }
 
+// NewMsgEditValidator creates a new MsgEditValidator instance and does sanity checks
+// on the given arguments before populating the message.
+func NewMsgEditValidator(args []interface{}) (*stakingtypes.MsgEditValidator, common.Address, error) {
+	if len(args) != 4 {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 4, len(args))
+	}
+
+	description := stakingtypes.Description{}
+	if descriptionInput, ok := args[0].(Description); ok {
+		description.Moniker = descriptionInput.Moniker
+		description.Identity = descriptionInput.Identity
+		description.Website = descriptionInput.Website
+		description.SecurityContact = descriptionInput.SecurityContact
+		description.Details = descriptionInput.Details
+	} else {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidDescription, args[0])
+	}
+
+	validatorAddress, ok := args[1].(string)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidType, "validatorAddress", "string", args[1])
+	}
+	valAddr, err := sdk.ValAddressFromBech32(validatorAddress)
+	if err != nil {
+		return nil, common.Address{}, err
+	}
+
+	commissionRateBigInt, ok := args[2].(*big.Int)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidType, "commissionRate", "int256", args[2])
+	}
+	commissionRate := new(sdk.Dec)
+	if commissionRateBigInt.Cmp(big.NewInt(DoNotModifyCommissionRate)) == 0 {
+		commissionRate = nil
+	} else {
+		*commissionRate = sdk.NewDecFromBigIntWithPrec(commissionRateBigInt, sdk.Precision)
+	}
+
+	minSelfDelegationBigInt, ok := args[3].(*big.Int)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidType, "minSelfDelegation", "int256", args[3])
+	}
+	minSelfDelegation := new(sdkmath.Int)
+	if minSelfDelegationBigInt.Cmp(big.NewInt(DoNotModifyMinSelfDelegation)) == 0 {
+		minSelfDelegation = nil
+	} else {
+		*minSelfDelegation = sdk.NewIntFromBigInt(minSelfDelegationBigInt)
+	}
```

Comment:
same, break lines.

Why do we need all these `if/else`?

Reply:
Because during the message execution phase, if the user does not want to modify CommissionRate or MinSelfDelegation, their value must be set to nil。See relevant code for details：
https://github.com/evmos/cosmos-sdk/blob/v0.47.5-evmos/x/staking/keeper/msg_server.go#L159-L183